### PR TITLE
vlm 70B+ in single card

### PR DESCRIPTION
**Pull request by n1ck-guo**

### CODE REVIEW

File: auto_round/mllm/template.py [hunk 1]
```diff
@@ -22,6 +22,15 @@
 from .processor import BasicProcessor, PROCESSORS
 
 TEMPLATES: Dict[str, "Template"] = {}
+SUPPORT_ONLY_TEXT_MODELS = [
+    "phi3_v",
+    "cogvlm2",
+    "llava",
+    "qwen2_vl",
+    "deepseek_vl_v2",
+    "chatglm",
```
Comment (wenhuach21) [inline auto_round/mllm/template.py]: no any postfix in chatglm?

File: auto_round/script/mllm.py [hunk 2]
```diff
@@ -270,7 +270,8 @@ def tune(args):
             os.environ["CUDA_VISIBLE_DEVICES"] = args.device
             args.device = ",".join(map(str, range(len(devices))))
             devices = args.device.replace(" ", "").split(',')
-        use_auto_mapping = True
+        if len(devices) > 1:
+            use_auto_mapping = True  ##for 70B model on single card, use auto will cause some layer offload to cpu
```
Comment (wenhuach21) [inline auto_round/script/mllm.py:274]: add ut for 70B

File: auto_round/mllm/autoround_mllm.py [hunk 3]
```diff
@@ -24,27 +24,47 @@
...
-def _only_text_test(model, tokenizer, device):
+def _only_text_test(model, tokenizer, device, model_type):
     """Test if the model whether can use text-only datasets."""
+
+    if model_type in SUPPORT_ONLY_TEXT_MODELS:  # save time
+        return True
+
+    device = detect_device(device)
+    text = ["only text", "test"]
+    tokenizer.padding_side = 'left'
```
Comment (wenhuach21) [inline auto_round/mllm/autoround_mllm.py]: it tokenizer has been changed here, the saved tokenizer should be changed too, right? That's not the expected behavior, the saved tokenizer should be exact the same as the original one.

Better revert the change after this thest

File: auto_round/mllm/template.py (hunk 1 above)
Comment (n1ck-guo) [reply auto_round/mllm/template.py]: No, only chatglm
https://huggingface.co/THUDM/glm-4v-9b/blob/6a1215eea198982bcfc2a639312471b9ac1ac044/config.json#L3

File: auto_round/mllm/template.py [hunk 4]
```diff
@@ -22,6 +22,15 @@
 from .processor import BasicProcessor, PROCESSORS
 
 TEMPLATES: Dict[str, "Template"] = {}
+SUPPORT_ONLY_TEXT_MODELS = [
+    "phi3_v",
+    "cogvlm2",
+    "llava",
```
Comment (wenhuach21) [inline auto_round/mllm/template.py]: should this variable be moved to special_model_handler.py?

File: auto_round/mllm/autoround_mllm.py [hunk 5]
```diff
@@ -41,8 +42,10 @@ def _only_text_test(model, tokenizer, device, model_type):
 
     device = detect_device(device)
     text = ["only text", "test"]
+    ori_padding_size = tokenizer.padding_side
     tokenizer.padding_side = 'left'
     inputs = tokenizer(text, return_tensors="pt", padding=True, truncation=True)
+    tokenizer.padding_size = ori_padding_size
```
Comment (wenhuach21) [inline auto_round/mllm/autoround_mllm.py]: side

File: auto_round/mllm/autoround_mllm.py [hunk 6]
```diff
@@ -41,8 +42,10 @@ def _only_text_test(model, tokenizer, device, model_type):
...
 
     if tokenizer.pad_token is None:
         tokenizer.pad_token = tokenizer.eos_token
```
Comment (wenhuach21) [inline auto_round/mllm/autoround_mllm.py]: revert

**Review by wenhuach21 — APPROVED** [review]
